hello-data-sidecars/hello-data-sidecar-superset/src/main/java/ch/bedag/dap/hellodata/sidecars/superset/service/dashboard/UploadDashboardsFileListener.java

```diff
@@ -0,0 +1,181 @@
+            log.debug("File {} created: {}", file.toPath(), created);
+        }
+        return uploadFolderPath;
+    }
+
+    /**
+     * Regex pattern replace all but numbers, letters, dashes and underscores.
+     */
+    private Path getUploadFolderPath(String filename) {
+        String uploadFolder = filename.replaceAll(FOLDER_NAMES_REGEX_PATTERN, "");
+        return Paths.get(System.getProperty("java.io.tmpdir"), "dashboards_upload", uploadFolder);
+    }
+
+    private void buildChunks(String binaryFileId, String filename, long totalChunks, long fileSize, Path destinationPath) throws IOException {
+        Path chunksFolderPath = getUploadFolderPath(binaryFileId);
+
+        if (!Files.exists(chunksFolderPath)) {
+            throw new RuntimeException("No chunks were found for filename : " + filename);
+        }
+
+        List<File> chunks = listChunks(chunksFolderPath);
+        if (chunks.isEmpty() || chunks.size() != totalChunks || validateChunkSize(fileSize, chunks)) {
+            throw new RuntimeException("The file is not supported. Upload canceled");
+        }
+        writeChunksToFile(destinationPath, chunks);
+        deleteTempBinaryFileData(binaryFileId);
+    }
+
+    private boolean validateChunkSize(long fileSize, List<File> chunks) {
+        return chunks.stream().mapToLong(File::length).sum() != fileSize;
+    }
+
+    private void deleteTempBinaryFileData(String filename) {
+        Path chunksFolderPath = getUploadFolderPath(filename);
+        if (Files.exists(chunksFolderPath)) {
+            List<File> chunks = listChunks(chunksFolderPath);
+            int nrFilesDeleted = 0;
+            for (File file : listChunks(chunksFolderPath)) {
+                if (file.delete()) {
+                    nrFilesDeleted++;
+                }
+            }
+
+            if (nrFilesDeleted == chunks.size()) {
+                try {
+                    FileUtils.deleteDirectory(new File(chunksFolderPath.toString()));
+                } catch (IOException e) {
+                    log.error("Chunks folder could not be deleted", e);
+                }
+            }
+        }
+    }
+
+    private List<File> listChunks(Path chunksFolderPath) {
+        File folderFile = new File(chunksFolderPath.toString());
+        File[] fileArray = folderFile.listFiles();
+        List<File> files = Arrays.asList(fileArray);
+        Collections.sort(files, (File o1, File o2) -> {
+            //remove extension
+            String chunkName1 = o1.getName().split("\\.")[0];
+            String chunkName2 = o2.getName().split("\\.")[0];
+            Long chunk1Number = Long.parseLong(chunkName1);
+            Long chunk2Number = Long.parseLong(chunkName2);
+            return chunk1Number.compareTo(chunk2Number);
+        });
+        return files;
+    }
+
+    private void writeChunksToFile(Path destinationPath, List<File> chunks) throws IOException {
+        destinationPath.getParent().toFile().mkdirs();
```

Comment:
I'd maybe extract the mkdirs-call out of this method, so the method only writes the chunks